api/api/models/audio.py

```diff
@@ -277,7 +276,7 @@ class DeletedAudio(AbstractDeletedMedia):
         primary_key=True,
         db_constraint=False,
         db_column="identifier",
-        related_name="deleted_audio",
+        related_name="deleted_media",
```

Comment:
Oops, these actually do not need to be defined anymore! I accidentally left these in when I was trying one of the other approaches I mentioned that still required manually configuring the related names. The metaclass approach intentionally removes the need for this.

Reply:
This is the approach that I was actually preferring over the metaclass because it is much more explicit and clearer.

Reply:
It doesn't solve the back-references to the class though. You can go through the related _set_ but if it's a one-to-one, there's no good way to do it, as far as I can tell, that won't cause an implicit query.

The complexity comes from trying to solve both problems:
- Make it possible to use generic references to instance relations
- Make it possible to use generic references to _model_s of those relations